Weka crashes on startup with Java 17

### Describe the bug
Weka crashes if using the _nixpkgs_ version

### Steps To Reproduce
Steps to reproduce the behavior:
1. Get weka from nixpkgs at f994293d1eb8812f032e8919e10a594567cf6ef7 (nixos-unstable)
2. Open weka

### Expected behavior
Weka opening that main window and working properly.

It neither opens the main window nor works properly.

### Screenshots
If applicable, add screenshots to help explain your problem.

### Additional context
Downloading the zip from [Weka's official site](https://sourceforge.net/projects/weka/) and running by using `java -jar weka.jar` from `openjdk11` works as expected.

```
lucasew@riverwood ~/WORKSPACE/TCC 0$ /nix/store/mhh9vz3rw8dj813y0q9fxykxpv7bykyi-weka-3.9.2/bin/weka 
java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @1c8a1bf1
	java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	weka.core.WekaPackageClassLoaderManager.injectClasses(WekaPackageClassLoaderManager.java:790)
	weka.core.WekaPackageClassLoaderManager.injectAllClassesInFromStream(WekaPackageClassLoaderManager.java:688)
	weka.core.WekaPackageClassLoaderManager.injectMTJCoreClasses(WekaPackageClassLoaderManager.java:118)
	weka.core.WekaPackageManager.<clinit>(WekaPackageManager.java:241)
	weka.core.ResourceUtils.readProperties(ResourceUtils.java:241)
	weka.core.ResourceUtils.readProperties(ResourceUtils.java:184)
	weka.core.Utils.readProperties(Utils.java:151)
	weka.core.logging.Logger.<clinit>(Logger.java:50)
	weka.gui.GUIChooserApp.main(GUIChooserApp.java:1607)
	java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.base/java.lang.reflect.Method.invoke(Method.java:568)
	weka.gui.SplashWindow.invokeMain(SplashWindow.java:306)
	weka.gui.GUIChooser.main(GUIChooser.java:92)

	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at weka.core.WekaPackageClassLoaderManager.injectClasses(WekaPackageClassLoaderManager.java:790)
	at weka.core.WekaPackageClassLoaderManager.injectAllClassesInFromStream(WekaPackageClassLoaderManager.java:688)
	at weka.core.WekaPackageClassLoaderManager.injectMTJCoreClasses(WekaPackageClassLoaderManager.java:118)
	at weka.core.WekaPackageManager.<clinit>(WekaPackageManager.java:241)
	at weka.core.ResourceUtils.readProperties(ResourceUtils.java:241)
	at weka.core.ResourceUtils.readProperties(ResourceUtils.java:184)
	at weka.core.Utils.readProperties(Utils.java:151)
	at weka.core.logging.Logger.<clinit>(Logger.java:50)
	at weka.gui.GUIChooserApp.main(GUIChooserApp.java:1607)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at weka.gui.SplashWindow.invokeMain(SplashWindow.java:306)
	at weka.gui.GUIChooser.main(GUIChooser.java:92)
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by weka.gui.GUIChooserApp (file:/nix/store/mhh9vz3rw8dj813y0q9fxykxpv7bykyi-weka-3.9.2/share/weka/weka.jar)
WARNING: Please consider reporting this to the maintainers of weka.gui.GUIChooserApp
WARNING: System::setSecurityManager will be removed in a future release
Exception in thread "main" java.lang.InternalError: Failed to invoke main method
	weka.gui.SplashWindow.invokeMain(SplashWindow.java:308)
	weka.gui.GUIChooser.main(GUIChooser.java:92)

	at weka.gui.SplashWindow.invokeMain(SplashWindow.java:308)
	at weka.gui.GUIChooser.main(GUIChooser.java:92)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at weka.gui.SplashWindow.invokeMain(SplashWindow.java:306)
	... 1 more
Caused by: java.lang.NoClassDefFoundError: no/uib/cipr/matrix/Matrix
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at weka.core.WekaPackageClassLoaderManager.forName(WekaPackageClassLoaderManager.java:198)
	at weka.core.WekaPackageClassLoaderManager.forName(WekaPackageClassLoaderManager.java:178)
	at weka.core.ClassDiscovery.find(ClassDiscovery.java:247)
	at weka.gui.GenericPropertiesCreator.generateOutputProperties(GenericPropertiesCreator.java:542)
	at weka.gui.GenericPropertiesCreator.execute(GenericPropertiesCreator.java:639)
	at weka.gui.GenericPropertiesCreator.<clinit>(GenericPropertiesCreator.java:167)
	at weka.core.WekaPackageManager.refreshGOEProperties(WekaPackageManager.java:1403)
	at weka.core.WekaPackageManager.loadPackages(WekaPackageManager.java:1328)
	at weka.core.WekaPackageManager.loadPackages(WekaPackageManager.java:1205)
	at weka.gui.GenericObjectEditor.determineClasses(GenericObjectEditor.java:172)
	at weka.gui.GenericObjectEditor.<clinit>(GenericObjectEditor.java:242)
	at weka.gui.GUIChooserApp.<init>(GUIChooserApp.java:662)
	at weka.gui.GUIChooserApp.createSingleton(GUIChooserApp.java:176)
	at weka.gui.GUIChooserApp.main(GUIChooserApp.java:1676)
	... 6 more
Caused by: java.lang.ClassNotFoundException: no.uib.cipr.matrix.Matrix
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 22 more
```

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -36054,7 +36054,7 @@ with pkgs;
 
   msieve = callPackage ../applications/science/math/msieve { };
 
-  weka = callPackage ../applications/science/math/weka { };
+  weka = callPackage ../applications/science/math/weka { jre = openjdk11; };
 
   yad = callPackage ../tools/misc/yad { };
 

